packages/next-playwright/src/next-fixture.ts

```diff
@@ -0,0 +1,64 @@
+import type { BrowserContext } from '@playwright/test'
+import type { NextWorkerFixture } from './next-worker-fixture'
+
+type MockFetch = (request: Request) => Response | Promise<Response>
+type MockFetchJson = (request: Request) => any | Promise<any>
+
+export interface NextFixture {
+  mockFetch: (url: string, mock: MockFetch) => void
+  mockFetchJson: (url: string, json: any | MockFetchJson) => void
+}
+
+class NextFixtureImpl implements NextFixture {
+  public testId: string
+  private worker: NextWorkerFixture
+
+  constructor(nextWorker: NextWorkerFixture) {
+    this.worker = nextWorker
+    this.testId = Math.random().toString(36).replace('0.', '')
+  }
+
+  teardown(): void {}
+
+  mockFetch(url: string, mock: MockFetch | Response | Promise<Response>): void {
+    const mockFunc = typeof mock === 'function' ? mock : () => mock
+    this.worker.mockFetch(this.testId, url, mockFunc)
+  }
+
+  mockFetchJson(url: string, json: any | ((request: Request) => any)): void {
+    this.worker.mockFetch(this.testId, url, async (request) => {
+      let value = json
+      if (typeof json === 'function') {
+        value = await Promise.resolve(json(request))
+      } else {
+        value = json
+      }
+      return new Response(JSON.stringify(value), {
+        status: 200,
+        headers: { 'Content-Type': 'application/json' },
+      })
+    })
+  }
+}
+
+export function setup({
+  nextWorker,
+  context,
+  extraHTTPHeaders,
+}: {
+  nextWorker: NextWorkerFixture
+  context: BrowserContext
+  extraHTTPHeaders: Record<string, string>
+}): NextFixture {
+  const fixture = new NextFixtureImpl(nextWorker)
+  context.setExtraHTTPHeaders({
+    ...extraHTTPHeaders,
+    'Next-Test-Proxy-Port': String(nextWorker.proxyPort),
+    'Next-Test-Data': fixture.testId,
+  })
+  return fixture
+}
+
+export function teardown(nextFixture: NextFixture): void {
+  ;(nextFixture as NextFixtureImpl).teardown()
```

Comment:
Can we improve the types here so that we don't need to cast?

Reply:
I just wanted to avoid leaking the `teardown` out of the main API.

Reply:
But, yeah, most likely can figure it out.